nixos/modules/system/activation/switch-to-configuration.py

```diff
@@ -0,0 +1,603 @@
+#! @python@
+
+from collections import namedtuple
+import os.path
+import re
+import signal
+import subprocess
+from syslog import openlog, syslog, LOG_NOTICE, LOG_ERR
+import sys
+from typing import Dict, NewType, List, Tuple
+
+out = "@out@"
+
+# To be robust against interruption, record what units need to be started etc.
+START_UNIT_FILE = "/run/systemd/start-list"
+RESTART_UNIT_FILE = "/run/systemd/restart-list"
+RELOAD_UNIT_FILE = "/run/systemd/reload-list"
+
+def print_err(*s, **kwargs):
+    print(*s, file=sys.stderr, **kwargs)
+
+def die_with_usage():
+    msg = """
+        Usage: {prog} [switch|boot|test]
+
+        switch:       make the configuration the boot default and activate now
+        boot:         make the configuration the boot default
+        test:         activate the configuration, but don\'t make it the boot default
+        dry-activate: show what would be done if this configuration were activated
+    """.format(prog=sys.argv[0])
+    print_err(msg)
+    sys.exit(1)
+
+def die(why: str):
```

Comment:
Functions should also ways have a return type if it its `None`. Otherwise the type check is skipped if a function does not have any typed arguments.